database/rrdcalctemplate.c

```diff
@@ -45,6 +45,11 @@ static int rrdcalctemplate_is_there_label_restriction(RRDCALCTEMPLATE *rt,  RRDH
 }
 
 static inline int rrdcalctemplate_test_additional_restriction(RRDCALCTEMPLATE *rt, RRDSET *st) {
+    if (rt->charts_pattern &&
+        !(simple_pattern_matches(rt->charts_pattern, st->id) ||
+         simple_pattern_matches(rt->charts_pattern, st->name)))
+        return 0;
```

Comment:
We need to think about this. Matching against both `id` and `name` makes it hard to write a filter.

For instance, see the screenshot

<img width="1107" alt="Screenshot 2021-05-19 at 17 46 27" src="https://user-images.githubusercontent.com/22274335/118833669-4e360600-b8ca-11eb-8b59-3073de26cf60.png">

As a user I expect the chart to be filtered out if I write `*netdata_child*`, but there is no guarantee it will do because `id` could be `...netdata-child...`.

Reply:
We are expecting to match either `id` or `name`, we do not expect both. :thinking:

Reply:
> We are expecting to match either

That is exactly the problem.

Reply:
ok, i am going to make a PR